engine/install/debian.md

```diff
@@ -93,7 +93,16 @@ from the repository.
     ```bash
     $ curl -fsSL {{ download-url-base }}/gpg | sudo apt-key add -
     ```
-
+    Newer versions of Debian have deprecated using `apt-key` like this. If you 
+    are receiving the following error:
+    ```
+    Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).
+    ```
+    then you can avoid it by using the following instead:
+    ```
+    curl -fsSL https://download.docker.com/linux/debian/gpg | sudo apt-key --keyring /etc/apt/trusted.gpg.d/docker-apt-key.gpg add -
```

Comment:
Does this also work on older versions? If so, we should probably replace the existing instructions to use the new one

This should probably also use the variable;

```suggestion
    curl -fsSL {{ download-url-base }}/gpg | sudo apt-key --keyring /etc/apt/trusted.gpg.d/docker-apt-key.gpg add -
```

Reply:
Good day @thaJeztah 

As per my [previous comment](https://github.com/docker/docker.github.io/pull/11974#issuecomment-751405853), this solution still rely on the deprecated `apt-key` tool. 

#11990 offers a Debian-compliant way of dealing with `apt-key` deprecation that applies to Debian 9 / Ubuntu 16.04 onward.

Thank you :)

Reply:
I believe the verification command needs also ` --keyring /etc/apt/trusted.gpg.d/docker-apt-key.gpg`, below.